client/pyscripts/fm11rf08_full.py

```diff
@@ -0,0 +1,942 @@
+def getPrefs():
+	global dpath
+
+	p.console("prefs show --json")
+	prefs = json.loads(p.grabbed_output)
+	dpath = prefs['file.default.dumppath'] + os.path.sep
+
+#+=============================================================================
+# Assert python version
+#==============================================================================
+def checkVer():
+	required_version = (3, 8)
+	if sys.version_info < required_version:
+		print(f"Python version: {sys.version}")
+		print(f"The script needs at least Python v{required_version[0]}.{required_version[1]}. Abort.")
+		exit()
+
+#+=============================================================================
+# Parse the CLi arguments
+# >> args.
+#==============================================================================
+def parseCli():
+	global args
+
+	parser = argparse.ArgumentParser(description='Full recovery of Fudan FM11RF08* cards.')
+
+	parser.add_argument('-n', '--nokeys',   action='store_true', help='extract data even if keys are missing')
+	parser.add_argument('-r', '--recover',  action='store_true', help='run key recovery script if required')
+	parser.add_argument('-f', '--force',    action='store_true', help='force recovery of keys')
+	parser.add_argument('-b', '--bambu',    action='store_true', help='force Bambu tag decode')
+	parser.add_argument('-v', '--validate', action='store_true', help='check Fudan signature (requires internet)')
+
+	args   = parser.parse_args()
+
+	if args.force == True:  args.recover = True
+
+#+=============================================================================
+# Find backdoor key
+# >> "dkey"
+# >> "blk0"
+'''
+[=]   # | sector 00 / 0x00                                | ascii
+[=] ----+-------------------------------------------------+-----------------
+[=]   0 | 5C B4 9C A6 D2 08 04 00 04 59 92 25 BF 5F 70 90 | \........Y.%._p.
+'''
+#==============================================================================
+def getDarkKey():
+	global dkey
+	global blk0
+
+	#          FM11RF08S        FM11RF08        FM11RF32
+	dklist = ["A396EFA4E24F", "A31667A8CEC1", "518b3354E760"]
+
+	print(prompt)
+	print(f"{prompt} Trying known backdoor keys...")
+
+	dkey = ""
+	for k in dklist:
+		cmd = f"hf mf rdbl -c 4 --key {k} --blk 0"
+		print(f"{prompt} `{cmd}`", end='', flush=True)
+		res = p.console(f"{cmd}")
+		if res == 0:
+			print(" - success")
+			dkey = k;
+			break;
+		print(f" - fail [{res}]")
+		_ = p.grabbed_output
+
+	if dkey == "":
+		print(f"{prompt}")
+		print(f"{prompt} ! Unknown key, or card not detected.")
+		exit(1)
+
+	for line in p.grabbed_output.split('\n'):
+		if " | " in line and "# | s" not in line:
+			blk0 = line[10:56+1]
+
+#+=============================================================================
+# Extract data from block 0
+# >> "uid"
+# >> "uids"
+#==============================================================================
+def decodeBlock0():
```

Comment:
it would be interesting to isolate this function in a separate file to be imported, so other ppl can use it in their scripts s well, or call the script with a provided block0

Reply:
If/when this script gets pulled, I would be flattered if functions were yanked to a library. If/when that happens it would probably be wise to pick a more "API" style name for the function. I dunno, maybe `hf_mf_block0_decode(binaryData[16])` ...but I'm sure you guys probably have a naming convention - so "whatever fits" is good by me

Same goes for the following 3 similar comments :)